File: 496-next-greater-element-i/496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> map = new HashMap<>();
        Stack<Integer> stack = new Stack<>();
        
        for(int i = 0; i<nums2.length; i++){
            while(!stack.isEmpty() && nums2[i] > stack.peek()){
                map.put(stack.pop(), nums2[i]);
            }
            stack.add(nums2[i]);
        }
        while(!stack.isEmpty()){
            map.put(stack.pop(), -1);
        }
        
        int[] res = new int[nums1.length];
        
        for(int i = 0; i< nums1.length; i++){
            res[i] = map.get(nums1[i]);
        }
        return res;
        
    }
}

File: 496-next-greater-element-i/496-next-greater-element-i.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function(nums1, nums2) {
    let myMap = new Map();
    for(let i = 0; i<nums2.length; i++){
        for(let j = i+1; j<nums2.length; j++){
            if(nums2[j] > nums2[i]){
                myMap.set(nums2[i], nums2[j])
                break;
            }else{
                myMap.set(nums2[i], -1);
            }
        }
    }
    myMap.set(nums2[nums2.length - 1], -1);
    let res = [];
    for(let i of nums1){
        if(myMap.has(i)){
            res.push(myMap.get(i));
        }
    }
    return res;
};